contracts/BSCValidatorSet.sol

```diff
@@ -479,7 +501,178 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
+    delete currentValidatorSetMap[validator];
+    // It is ok that the validatorSet is not in order.
+    if (index != currentValidatorSet.length - 1) {
+      currentValidatorSet[index] = currentValidatorSet[currentValidatorSet.length - 1];
+      currentValidatorSetMap[currentValidatorSet[index].consensusAddress] = index + 1;
+    }
+    currentValidatorSet.pop();
+  }
+
+  function _forceMaintainingValidatorsExit(Validator[] memory validatorSet) private returns (Validator[] memory unjailedValidatorSet){
+    uint256 numOfFelony = 0;
+    address validator;
+    bool isFelony;
+
+    // 1. validators exit maintenance
+    for (uint i = 0; i < maintainingValidatorSet.length; i++) {
+      validator = maintainingValidatorSet[i].consensusAddress;
+      // exit maintenance
+      isFelony = _exitMaintenance(validator);
+      delete maintainInfoMap[validator];
+
+      if (!isFelony || numOfFelony >= validatorSet.length - 1) {
+        continue;
+      }
+
+      // record the jailed validator in validatorSet
+      for (uint index = 0; index < validatorSet.length; index++) {
+        if (validatorSet[index].consensusAddress == validator) {
+          validatorSet[index].jailed = true;
+          numOfFelony++;
+          break;
+        }
+      }
+    }
+
+    // 2. clear all maintain info
+    delete maintainingValidatorSet;
+
+    // 3. get unjailed validators from validatorSet
+    unjailedValidatorSet = new Validator[](validatorSet.length - numOfFelony);
+    uint256 i = 0;
+    for (uint index = 0; index < validatorSet.length; index++) {
+      if (!validatorSet[index].jailed) {
+        unjailedValidatorSet[i] = validatorSet[index];
+        i++;
+      }
+    }
+
+    return unjailedValidatorSet;
+  }
+
+  function _enterMaintenance(address validator, uint256 index) private {
+    // step 1: modify status of the validator
+    MaintainInfo storage maintainInfo = maintainInfoMap[validator];
+    maintainInfo.isMaintaining = true;
+    maintainInfo.startBlockNumber = block.number;
+
+    // step 2: add the validator to maintainingValidatorSet
+    maintainingValidatorSet.push(currentValidatorSet[index]);
+    maintainInfo.index = maintainingValidatorSet.length - 1;
+
+    // step 3: remove the validator from currentValidatorSet
+    _removeFromCurrentValidatorSet(validator, index);
+
+    emit validatorMaintain(validator);
+  }
+
+  function _exitMaintenance(address validator) private returns (bool isFelony){
+    // step 1: modify status of the validator
+    MaintainInfo storage maintainInfo = maintainInfoMap[validator];
+    maintainInfo.isMaintaining = false;
+
+    // step 2: add the validator to currentValidatorSet
+    uint256 index = maintainInfo.index;
+    currentValidatorSet.push(maintainingValidatorSet[index]);
+    currentValidatorSetMap[validator] = currentValidatorSet.length;
+
+    // step 3: remove the validator from maintainingValidatorSet
```

Comment:
Shall we remove it? otherwise, the validator can enter maintainence multi times within one day.

Reply:
If the validator entered maintenance, `|| maintainInfoMap[validator].startBlockNumber > 0`, this check condition will not pass, then the validator can only enter maintenance once within one day